eox_nelp/stats/tests/tests_metrics.py

```diff
@@ -0,0 +1,371 @@
+        chapter.get_children.return_value = sequentials
+        chapters = [
+            chapter,
+            chapter,
+            chapter,
+            chapter,
+        ]
+        # Prepare course
+        course = Mock()
+        course.display_name = "testing"
+        course.get_children.return_value = chapters
+        # Set course
+        self.modulestore.return_value.get_course.return_value = course
+
+        # this block set the CourseEnrollment mock and its returned values.
+        filter_result = CourseEnrollment.objects.filter.return_value
+        values_result = filter_result.values.return_value
+        distinct_result = values_result.distinct.return_value
+        distinct_result.count.return_value = 5874
+
+        # this block set the CourseAccessRole mock and its returned values.
+        filter_result = CourseAccessRole.objects.filter.return_value
+        values_result = filter_result.values.return_value
+        distinct_result = values_result.distinct.return_value
+        distinct_result.count.return_value = 5
+
+    def tearDown(self):
+        """Clean cache and restarts mocks"""
+        # This line just verifies that de get_course modulestore method cwas called with the right parameter
+        self.modulestore.return_value.get_course.assert_called_once_with(self.course_key)
+
+        CourseAccessRole.reset_mock()
+        CourseEnrollment.reset_mock()
+        cache.clear()
+        self.patch.stop()
+
+    def test_get_right_id(self):
+        """Based on the initial conditions, this check that the course metrics has the expected id.
+
+        Expected behavior:
+            - 'id' value is the expected.
+        """
+        course = get_course_metrics(self.course_key)
+
+        self.assertEqual(str(self.course_key), course["id"])
+
+    def test_get_right_name(self):
+        """Setting a different course display name, this verifies that the course metrics has the same name.
+
+        Expected behavior:
+            - 'name' value is the expected
+        """
+        self.modulestore.return_value.get_course.return_value.display_name = "Amazing course"
+
+        course = get_course_metrics(self.course_key)
+
+        self.assertEqual("Amazing course", course["name"])
+
+    def test_get_right_learners_metric(self):
+        """Based on the initial conditions, this check that the course metrics has the expected learners value.
+
+        Expected behavior:
+            - 'learners' value is the expected
+        """
+        course = get_course_metrics(self.course_key)
+
+        self.assertEqual(5874, course["learners"])
+
+    def test_get_right_instructors_metric(self):
+        """Based on the initial conditions, this check that the course metrics has the expected instructors value.
+
+        Expected behavior:
+            - 'instructors' value is the expected
+        """
+        course = get_course_metrics(self.course_key)
+
+        self.assertEqual(5, course["instructors"])
```

Comment:
This one could be compared with this. https://github.com/eduNEXT/eox-nelp/pull/55/files#diff-ed3b504ce4a13370c20a7d619e584c45fee7c0d005ca4a6abc4974cb05ec1a04R262
but that var is used twice.
Like above  the rename of var could help.